3rdparty/dlmalloc/allocator.c

```diff
@@ -115,3 +116,18 @@ size_t oe_allocator_malloc_usable_size(void* ptr)
 {
     return dlmalloc_usable_size(ptr);
 }
+
+int oe_allocator_mallinfo(oe_mallinfo_t* info)
+{
+    info->max_total_heap_size = _heap_end - _heap_start;
```

Comment:
Can _heap_end ever decrease? If so, we'd have to maintain a running max heap size.

Reply:
As suggested on #3484, not currently, no.

Reply:
I've split `_max_total_heap` for clarity, as suggested by @mjp41 on the other PR.